core/services/log/broadcaster.go

```diff
@@ -163,9 +174,14 @@ func (b *broadcaster) awaitInitialSubscribers() {
 }
 
 func (b *broadcaster) Register(listener Listener, opts ListenerOpts) (unsubscribe func()) {
-	if len(opts.Logs) < 1 {
+	if len(opts.Logs) == 0 && len(opts.LogsWithTopics) == 0 {
 		logger.Fatal("Must supply at least 1 Log to Register")
 	}
+
+	if len(opts.Logs) > 0 && len(opts.LogsWithTopics) > 0 {
+		logger.Fatal("Must use either Logs or LogsWithTopics but not both")
```

Comment:
Will this crash the app on a bad job spec? Or must it necessarily be a programming error?

Reply:
yeah always a programming error - caller to the Register method provides one of these fields in the code